pkg/core/workflow.go

```diff
@@ -82,20 +82,23 @@ const (
 
 // Node defines a single step of a workflow.
 type Node struct {
-	Name     string        `json:"name"`
-	Type     NodeType      `json:"type"`
-	State    NodeState     `json:"state"`
-	Serial   *NodeSerial   `json:"serial,omitempty"`
-	Parallel *NodeParallel `json:"parallel,omitempty"`
-	Template string        `json:"template"`
-	UID      string        `json:"uid"`
+	Name                string              `json:"name"`
+	Type                NodeType            `json:"type"`
+	State               NodeState           `json:"state"`
+	Serial              *NodeSerial         `json:"serial,omitempty"`
+	Parallel            *NodeParallel       `json:"parallel,omitempty"`
+	ConditionalBranches []ConditionalBranch `json:"conditional_branches,omitempty"`
+	Template            string              `json:"template"`
+	UID                 string              `json:"uid"`
 }
 
 type NodeNameWithTemplate struct {
 	Name     string `json:"name,omitempty"`
 	Template string `json:"template,omitempty"`
 }
 
+// FIXME: unnecessary another wrap of []NodeNameWithTemplate, like struct NodeSerial and NodeParallel
```

Comment:
Could u explain more about this? I'm not sure why it's unnecessary.

Reply:
I have already approved because this comment won't affect the changes.

Reply:
related issue https://github.com/chaos-mesh/chaos-mesh/issues/2054, struct `NodeSerial` and `NodeParallel` only contains one field `[]NodeNameWithTemplate`, it looks ugly. :cry:

Reply:
Got it.